packages/eslint-plugin/tests/rules/strict-boolean-expressions.test.ts

```diff
@@ -965,6 +984,56 @@ if (y) {
       ],
     }),
 
+    // nullable enum in boolean context
+    {
+      options: [{ allowNullableEnum: false }],
+      code: `
+        enum ExampleEnum {
+          This = 0,
+          That = 1,
+        }
+        const rand = Math.random();
+        let theEnum: ExampleEnum | null = null;
+        if (rand < 0.3) {
+          theEnum = ExampleEnum.This;
+        }
+        if (theEnum) {
+        }
+        if (!theEnum) {
+        }
+      `,
```

Comment:
We generally ask that each test case tests a single area of logic. That makes it easier to debug them in isolation, and if one facet of logic fails, we can see exactly which case/test(s) it impacts. Could you please split both the `valid` and `invalid` cases up into the two error cases they're each meant to test?

Reply:
splitted into two error cases!
https://github.com/typescript-eslint/typescript-eslint/pull/6096/commits/fae50f4ae3ce53bf0bf46ac57c395c3bc52ab7ff